src/main/kotlin/no/nav/tilleggsstonader/sak/vilkår/vilkårperiode/domain/Vilkårperiode.kt

```diff
@@ -48,10 +46,27 @@ data class Vilkårperiode(
         }
 
         validerBegrunnelseNedsattArbeidsevne()
+        validerBegrunnelseIngenAktivitetEllerMålgruppe()
 
         validerSlettefelter()
     }
 
+    private fun validerAktivitetsdager() {
+        if (type is AktivitetType) {
+            if (type === AktivitetType.INGEN_AKTIVITET) {
+                brukerfeilHvis(aktivitetsdager != null) { "Kan ikke registrere aktivitetsdager på ingen aktivitet" }
+            } else {
+                brukerfeilHvis(aktivitetsdager !in 1..5) {
+                    "Aktivitetsdager må være et heltall mellom 1 og 5"
+                }
+            }
+        }
+
+        if (type is MålgruppeType) {
+            brukerfeilHvis(aktivitetsdager !== null) { "Kan ikke registrere aktivitetsdager på målgruppe" }
```

Comment:
Klassisk eksempel på hva som skjer når man tvinger to ting som _ikke egentlig er det samme_ (Aktivitet, Målgruppe), inn i en felles abstraksjon (Vilkårsperioder) 😅

Reply:
Ja enig, det er en tanke å splitte de fra hverandre men det er noe større som bør gjøres som en egen oppgave og som et samarbeid mellom flere for å sørge for at vi får en modell som er bra 😢